scp.c

```diff
@@ -1387,8 +1429,18 @@ rsource(char *name, struct stat *statp)
 			return;
 		}
 	}
-	(void) snprintf(path, sizeof path, "D%04o %d %.1024s\n",
-	    (u_int) (statp->st_mode & FILEMODEMASK), 0, last);
+
+#ifdef WINDOWS
+	while (len = snprintf(path, sizeof path, "D%04o %d %.1024s\n",
+		  (u_int)(statp->st_mode & FILEMODEMASK), 0, last) >= path_len) {
+		path = xreallocarray(path, len + 1, sizeof(char));
```

Comment:
We wouldn't come here at all because snprintf writes at max sizeof(path) i.e., path_len?

Reply:
Have you checked this change by having a file / folder with more than 260 characters?

Reply:
snprintf returns the number of characters that would have been written if "path" was large enough to fit the whole string in that case. So we can compare path_len to the return value of sprintf to identify if we need to increase the size of "path".